packages/sanity/src/structure/components/RenderActionCollectionState.tsx

```diff
@@ -41,18 +41,20 @@ export const RenderActionCollectionState = memo((props: RenderActionCollectionPr
       args={actionProps}
       group={group}
     >
-      {({states}) => (
-        <ActionsGuardWrapper
-          states={states}
-          documentId={
-            actionProps.liveEditSchemaType
-              ? getPublishedId(actionProps.id)
-              : getDraftId(actionProps.id)
-          }
-        >
-          {children}
-        </ActionsGuardWrapper>
-      )}
+      {({states}) => {
+        return (
+          <ActionsGuardWrapper
+            states={states}
+            documentId={
+              actionProps.liveEditSchemaType
```

Comment:
Shouldnt this be using the getDocumentIdFromDocumentActionProps helper? This looks like the same code, but maybe im seeing it wrong